<launch>
	
	<arg name="filepath" default="/home/luiz/catkin_ws/src/bboat_pkg/mission_files/sinus_ensta.txt" />
	<arg name="mission_simu" default="1" />

	<arg name="plot" default="1" />
	
	<!-- Setting ROS control -->
	<node name="set_sysid"
	pkg="mavros"
	type="mavparam"
	args="set SYSID_MYGCS 1"
	output="screen"
	/>

	<!-- BBoat Simu node -->
	<node name="bboat_simu_node"
	type="simu_node.py"
	pkg="bboat_pkg"
	output="screen"
	/>

	<!-- BBoat Mission node -->
	<node name="bboat_mission_node"
	type="mission_node.py"
	pkg="bboat_pkg"
	output="screen">
		<param name="filepath" value="$(arg filepath)" />
		<param name="mission_simu" value="$(arg mission_simu)" />
	</node>

	<!-- BBoat Virtual Sail Boat node -->
	<!--node name="bboat_vsb_node"
	type="virtual_sb_node.py"
	pkg="bboat_pkg"
	output="screen"
	/ -->

	<!-- BBoat Controller node -->
	<node name="bboat_controller_node"
	type="controller_node.py"
	pkg="bboat_pkg"
	output="screen"
	/>

	<!-- Sensor node -->
	<node name="bboat_sensor_node"
	type="sensor_node.py"
	pkg="bboat_pkg"
	output="screen"
	/>

	<!-- Joystick node -->
	<!-- <node name="joystick_node"
	type="joy_node"
	pkg="joy"
	output="screen"
	/> -->
	
	<!-- BBoat Plotter node -->
	<node name="bboat_plotter_node"
	type="plotter_node.py"
	pkg="bboat_pkg"
	output="screen">
		<param name="plot" value="$(arg plot)" />
		<param name="mission_simu" value="$(arg mission_simu)" />
	</node>
	
</launch>
